Tasks/VsTest/distributedtest.ts

```diff
@@ -59,68 +60,169 @@ export class DistributedTest {
+            inputDataContract['COR_PROFILER_PATH_32'] = this.dtaTestConfig.toolsInstallerConfig.x86ProfilerProxyDLLLocation;
+            inputDataContract['COR_PROFILER_PATH_64'] = this.dtaTestConfig.toolsInstallerConfig.x64ProfilerProxyDLLLocation;
+            inputDataContract['ForcePlatformV2'] = true;
         }
 
         if (utils.Helper.isDebugEnabled()) {
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.EnableDiagLogs', 'true');
+            inputDataContract['Logging']['DebugLogging'] = true;
         }
 
         if (this.dtaTestConfig.pathtoCustomTestAdapters) {
             const testAdapters = tl.findMatch(this.dtaTestConfig.pathtoCustomTestAdapters, '**\\*TestAdapter.dll');
             if (!testAdapters || (testAdapters && testAdapters.length === 0)) {
                 tl.warning(tl.loc('pathToCustomAdaptersContainsNoAdapters', this.dtaTestConfig.pathtoCustomTestAdapters));
             }
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.CustomTestAdapters', this.dtaTestConfig.pathtoCustomTestAdapters);
+            inputDataContract['ExecutionSettings']['CustomTestAdapters'] =  this.dtaTestConfig.pathtoCustomTestAdapters;
         }
 
         // Set proxy settings to environment if provided
         if (!utils.Helper.isNullEmptyOrUndefined(this.dtaTestConfig.proxyConfiguration.proxyUrl)) {
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyUrl', this.dtaTestConfig.proxyConfiguration.proxyUrl);
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyUsername', this.dtaTestConfig.proxyConfiguration.proxyUserName);
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyPassword', this.dtaTestConfig.proxyConfiguration.proxyPassword);
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.ProxyBypassHosts', this.dtaTestConfig.proxyConfiguration.proxyBypassHosts);
+            inputDataContract['ProxySettings']['ProxyUrl'] = this.dtaTestConfig.proxyConfiguration.proxyUrl;
+            inputDataContract['ProxySettings']['ProxyUsername'] = this.dtaTestConfig.proxyConfiguration.proxyUserName;
+            inputDataContract['ProxySettings']['ProxyPassword'] =  this.dtaTestConfig.proxyConfiguration.proxyPassword;
+            inputDataContract['ProxySettings']['ProxyBypassHosts'] = this.dtaTestConfig.proxyConfiguration.proxyBypassHosts;
+        }
+
+        inputDataContract['TestSelectionSettings']['AssemblyBasedTestSelection']['SourceFilter'] = this.dtaTestConfig.sourceFilter.join('|');
+
+        let settingsFile = this.dtaTestConfig.settingsFile;
+        if (utils.Helper.pathExistsAsFile(settingsFile)) {
+            tl.debug('Final runsettings file being used:');
+            utils.Helper.readFileContents(settingsFile, 'utf-8').then(function (settings) {
+                tl.debug('Running VsTest with settings : ');
+                utils.Helper.printMultiLineLog(settings, (logLine) => { console.log('##vso[task.debug]' + logLine); });
+            });
+        }
+
```

Comment:
i will not be accepting to enabling automatic rerun/tia getting invoked because certain properties/nodes exist in json. Customer should be clear on what he/she is enabling/disabling. it should be similar to task UI. if we develop simple UI tool for generation of the same, it should be on par with CI/CD UI experience.

Reply:
@cltshivash, @prawalagarwal, @acesiddhu whats are final call on this?

Reply:
We have decided to close this review with the existing design and take this as feedback for the next phase/PR. This should be closed before the next PR.